Model/OrderData.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Metrilo\Analytics\Model;
+
+class OrderData
+{
+    public function __construct(
+        \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderCollection,
+        \Magento\Directory\Api\CountryInformationAcquirerInterface $countryInterface
+    ) {
+        $this->orderCollection = $orderCollection;
+        $this->countryInterface = $countryInterface;
+    }
+
+    public function getOrders($storeId)
+    {
+        $ordersArray = [];
+        $orders = $this->getOrderQuery($storeId);
+
+        foreach ($orders as $order) {
+            if(!$order->getCustomerEmail()) {
+                continue;
+            }
+
+            $orderItems = $order->getAllItems(); // getAllVisibleItems() returns only parent products (for configurables) from order BUT ignoring child product quantities for BUNDLE products
+
+            foreach ($orderItems as $orderItem) {
+                $itemType = $orderItem->getProductType();
+                if ($itemType != 'configurable' && $itemType != 'bundle') { // exclude configurable/bundle parent product returned by getAllItems() method
+                    $orderProducts[] = [
+                        'productId' => $orderItem->getProductId(),
+                        'quantity'  => $orderItem->getQtyOrdered()
+                    ];
+                }
+            }
+
+            $orderBillingData = $order->getBillingAddress();
+            $countryData      = $this->countryInterface->getCountryInfo($orderBillingData->getCountryId());
+            $street           = $orderBillingData->getStreet();
+            $couponCode       = $order->getCouponCode();
+
+            $orderBilling = [
+                "firstName"     => $orderBillingData->getFirstname(),
+                "lastName"      => $orderBillingData->getLastname(),
+                "address"       => is_array($street) ? implode(PHP_EOL, $street) : $street,
+                "city"          => $orderBillingData->getCity(),
+                "country"       => $countryData->getFullNameEnglish(),
```

Comment:
second time is a charm... why do you treat `city` and `country` attributes differently?

Reply:
Country name is not present in order billing address object, instead there is countryID by witch country name can be extracted using \Magento\Directory\Api\CountryInformationAcquirerInterface